### PR TITLE
Followup of #4784 - fixes remaining issues of #4762

### PR DESCRIPTION
**Pull request by matthiasgeiger**
Fixes the remaining problems mentioned in https://github.com/JabRef/jabref/pull/4784#pullrequestreview-216419250:
- table is now working again in case a new file is opened (also fixes https://github.com/JabRef/jabref/issues/4793)
- in case preferences are saved entry editor is working properly again
- external changes are now also detected in case file name has changed due to "Save As"

-- 
- ~~Change in CHANGELOG.md described~~
- ~~Tests created for changes~~
- [x] Manually tested changed features in running JabRef
- ~~Screenshots added in PR description (for bigger UI changes)~~
- [x] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- ~~Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)~~

### CODE REVIEW

File: src/main/java/org/jabref/gui/BasePanel.java [hunk 1]
```diff
@@ -790,6 +774,22 @@ public void setupMainPanel() {
         dividerPositionSubscription = EasyBind.monadic(Bindings.valueAt(splitPane.getDividers(), 0))
                                               .flatMap(SplitPane.Divider::positionProperty)
                                               .subscribe((observable, oldValue, newValue) -> saveDividerLocation(newValue));
+
+        // Add changePane in case a file is present - otherwise just add the splitPane to the panel
+        Optional<File> file = bibDatabaseContext.getDatabaseFile();
+        if (file.isPresent()) {
+            // create changeMonitor and changePane so we get notifications about outside changes to the file.
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/BasePanel.java]: Please use Path class, there should be a method getDatabasePath

**Comment by Siedlerchr** [comment]
Thanks for the followup! Works fine now!

**Comment by matthiasgeiger** [comment]
Thank you for the Path fix 😉

**Comment by tobiasdiez** [comment]
Thanks for fixing my mess! Much appreciated ;-)

**Pull request merged**